src/Native/System.Native/pal_io.h

```diff
@@ -22,6 +22,16 @@ struct FileStatus
     int64_t BirthTime; // time the file was created
 };
 
+
+/************
+ * The enums below are, unless otherwise specified, common values on the platforms
+ * we currently support. These values are not guaranteed by any standard, unless
+ * called out specifically, and could differ on other platforms. When adding new
+ * enums and values, be sure to add a static_assert in the implementation file
+ * to ensure your constant is as expected on all platforms.
+ */
+
```

Comment:
This isn't quite what I'd want the header to say. The values below in the header are fixed and correct for managed callers to use forever. We must never change them. The implementation must either: static_assert that they are equal to the native equivalent OR convert them appropriately.